UefiTestingPkg/FunctionalSystemTests/MemoryProtectionTest/App/MemoryProtectionTestApp.c

```diff
@@ -1052,46 +1131,90 @@ UefiCpuStackGuard (
   )
 {
   MEMORY_PROTECTION_TEST_CONTEXT  MemoryProtectionContext = (*(MEMORY_PROTECTION_TEST_CONTEXT *)Context);
+  EFI_PHYSICAL_ADDRESS            StackBase;
+  EFI_PEI_HOB_POINTERS            Hob;
+  EFI_HOB_MEMORY_ALLOCATION       *MemoryHob;
+  UINT64                          Attributes;
+  LIST_ENTRY                      *List;
+  CPU_MP_DEBUG_PROTOCOL           *Entry;
 
   DEBUG ((DEBUG_INFO, "%a - Testing CPU Stack Guard\n", __FUNCTION__));
 
-  if (MemoryProtectionContext.DynamicActive) {
-    if (mNonstopModeProtocol == NULL) {
-      UT_ASSERT_NOT_EFI_ERROR (GetNonstopProtocol ());
+  if (MemoryProtectionContext.TestingMethod == MemoryProtectionTestMemoryAttributeProtocol) {
+    UT_ASSERT_NOT_NULL (mMemoryAttributeProtocol);
+
+    StackBase = 0;
+    Hob.Raw   = GetHobList ();
+    while ((Hob.Raw = GetNextHob (EFI_HOB_TYPE_MEMORY_ALLOCATION, Hob.Raw)) != NULL) {
+      MemoryHob = Hob.MemoryAllocation;
+      if (CompareGuid (&gEfiHobMemoryAllocStackGuid, &MemoryHob->AllocDescriptor.Name)) {
+        StackBase = MemoryHob->AllocDescriptor.MemoryBaseAddress;
+        UT_ASSERT_EQUAL (StackBase & EFI_PAGE_MASK, 0);
+        break;
+      }
+
+      Hob.Raw = GET_NEXT_HOB (Hob);
     }
 
+    UT_ASSERT_NOT_EQUAL (StackBase, 0);

```

Comment:
`UT_ASSERT_EQUAL (StackBase & EFI_PAGE_MASK, 0);` is inside the HOB search loop. Should this line (`UT_ASSERT_NOT_EQUAL (StackBase, 0);`) precede that? Then, you'd exit on a non-zero, page-aligned stack into the memory attributes check.

Reply:
I initialize StackBase to zero and use this line to ensure that the HOB search routine found the stack.